src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/freestyle/FreeStylePostBuildStep.java

```diff
@@ -75,9 +75,9 @@ public boolean perform(
 
         final FreestyleBuildInfoRequest request =
                 new FreestyleBuildInfoRequest(this.getSite(), this.branch, build);
-        JiraSenderFactory.getInstance(pipelineLogger)
+        new JiraSenderFactory(pipelineLogger)
                 .getFreestyleBuildInfoSender()
-                .sendBuildInfo(request);
+                .sendBuildInfo(request, pipelineLogger);
```

Comment:
if you've already gave the pipeline logger in the constructor, why give it again for the `sendBuildInfo`?

Reply:
Good call. I actually now went and removed the `PipelineLogger` as a field everywhere and it's now just being passed through the method calls. Much better passing this around because otherwise it might get mixed up and we're logging into a stream that we don't want to log into...

Reply:
Since this PR is not using the `JiraSenderFactory.getInstance` not, I'd like to understand why is it "Much better passing this around"

Reply:
I've changed it back so it uses `JiraSenderFactory.getInstance()` again. Initially, I thought it would be a good idea to add the `PipelineLogger` as an instance field to the `JiraSenderFactory` so it can be re-used in all the "senders" the factory provides, but then I realized that sharing a `PipelineLogger` instance isn't a good idea, so I reverted that and am now passing the `PipelineLogger` instance around in each method call that needs to do some logging.